portal-bridge/src/cli.rs

```diff
@@ -154,6 +159,76 @@ pub struct BridgeConfig {
         help = "The maximum number of concurrent offer rpc requests for state bridge."
     )]
     pub offer_limit: usize,
+
+    #[arg(
+        default_value_t = ENR_OFFER_LIMIT,
+        long = "enr-offer-limit",
+        help = "The maximum number of enrs to gossip per content key (STATE BRIDGE ONLY)."
+    )]
+    pub enr_offer_limit: usize,
+
+    #[arg(
+        long = "filter-clients",
+        help = "Filter clients out from offer request (STATE BRIDGE ONLY).",
+        value_parser = client_parser,
+        default_value = "",
+    )]
+    pub filter_clients: Arc<Vec<ClientType>>,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum ClientType {
+    Fluffy,
+    Trin,
+    Shisui,
+    Ultralight,
+    Unknown,
+}
+
+impl TryFrom<&str> for ClientType {
+    type Error = String;
+
+    fn try_from(s: &str) -> Result<Self, Self::Error> {
+        match s {
+            "fluffy" => Ok(ClientType::Fluffy),
+            "trin" => Ok(ClientType::Trin),
+            "shisui" => Ok(ClientType::Shisui),
+            "ultralight" => Ok(ClientType::Ultralight),
+            "unknown" => Ok(ClientType::Unknown),
+            _ => Err(format!("Unknown client type: {s}")),
+        }
+    }
+}
+
+impl From<&Enr> for ClientType {
+    fn from(enr: &Enr) -> Self {
+        if let Some(client_id) = UtpEnr(enr.clone()).client() {
```

Comment:
Since this doesn't have anything to do with uTP, this usage stands out as strange. Should it be renamed? Should we extract enr parsing logic into a new more-generic struct?

Reply:
Enr's shouldn't hold client_id in a month or two, so whatever is chosen it shouldn't be the worst

Reply:
Now we're really going on a tangent, but since we started: I can see some downsides to this direction of pinging for metadata, like we would have a new funny class of peer that is connected, but we don't have enough information to interact with them yet.

Reply:
> but we don't have enough information to interact with them yet.

I am not sure what this means. We shouldn't need to know which implementation we are talking to interop with other clients.

Reply:
> We shouldn't need to know which implementation we are talking to interop with other clients.

We shouldn't, but clearly it is the most expedient option at times, like in this PR. I'm all for changing to eventually doing peer scoring and all that. It will just be more painful to use this PRs approach until then (and maybe there are other similar scenarios yet to be discovered).

Reply:
Agreed using `UtpEnr` here doesn't track that well. `Enr` is a "foreign" type, so it's not so simple to implement `client()` directly on it. I just copied the logic over since this might be changed soon, but regardless of where we go with ping's in the future, we'll be able to recognize that this logic also needs updating if we remove `ENR_PORTAL_CLIENT_KEY`